File: Algorithms/334/increasing-triplet-subsequence.cpp
class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        if (nums.size() < 3) return false;
        
        int dp[3];
        memset(dp, 0x7F, sizeof dp);
        
        for (int num : nums) {
            for (int i = 0; i < 3; ++i) {
                if (dp[i] < num) continue;
                if (i == 2) return true;
                dp[i] = num;
                break;
            }
        }
        
        return false;
    }
};

File: Algorithms/334/increasing-triplet-subsequence.swift
class Solution {
    func increasingTriplet(_ nums: [Int]) -> Bool {
        if nums.count < 3 {
            return false
        }
        
        var dp = Array.init(repeating: 0x7FFFFFFF, count: 3)
        for num in nums {
            for i in 0..<dp.count {
                if dp[i] < num {
                    continue
                }
                if i == 2 {
                    return true
                }
                dp[i] = num
                break;
            }
        }
        
        return false
    }
}